### ISSUE TITLE
HTML5 compiler issue - code not working

### ISSUE THREAD

**Issue by rwkay** [YoYoGames/GameMaker-Bugs#4725]
### Description

```
function Thingo() constructor {
    blah = function(a) {
        show_message(self)
        show_message(typeof(self))
        self[$ a] = "hi";
    } 
}

var thingo = new Thingo();
thingo.blah("kaboom");
```
[Bug-4725.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14279792/Bug-4725.zip)

Code is not working correctly on HTML5 compiler

### Expected Change

it should work properly.

### Steps To Reproduce

1. Start GameMaker
2. 
3. 
4. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.200 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by Sidorakh** [comment]
[struct-issue.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/14278965/struct-issue.zip)
Attached a project that actually shows this issue, my initial assumption about the cause was wrong

**Comment by rwkay** [comment]
Fixed in 2024.2

**Comment by sihammill** [comment]
Fix verified in IDE v2024.200.0.497  Runtime v2024.200.0.514
